addons/mail/static/src/web/messaging_menu/messaging_menu.xml

```diff
@@ -74,28 +74,29 @@
                     onSwipeRight="hasTouch() ? { action: () => this.cancelNotifications(failure), icon: 'fa-times-circle', bgColor: 'bg-warning' } : undefined"
                 />
             </t>
-            <t t-foreach="displayedPreviews" t-as="preview" t-key="preview.id">
+            <t t-foreach="displayedPreviews" t-as="preview" t-key="preview.localId">
+                <t t-set="message" t-value="preview.is_pinned ? preview.newestMessage : preview.newestNeedactionMessage"/>
                 <NotificationItem
-                    body="preview.message?.inlineBody"
-                    count="preview.count"
+                    body="message?.inlineBody"
+                    counter="preview.type === 'chat' ? preview.message_unread_counter : preview.hasNeedactionMessages ? preview.message_needaction_counter : undefined"
```

Comment:
everything that applies to `chat` should also apply to `group`, and possibly also `livechat`